src/main/java/uk/gov/indexer/ctserver/CTEntryLeaf.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.indexer.ctserver;
+
+import java.util.Arrays;
+import java.util.Base64;
+
+@SuppressWarnings({"FieldCanBeLocal", "unused"})
+public class CTEntryLeaf {
+    public final byte version;
+
+    public final byte merkleLeafType;
+
+    public final byte[] timetamp;
+
+    public final byte[] entryType;
+
+    public final byte[] contentLength;
+
+    public final byte[] payload;
+
+    public final byte[] sctExtension;
+
+    public CTEntryLeaf(String leafHash) {
+        byte[] decodeBytes = Base64.getDecoder().decode(leafHash);
+
+        int decodeBytesLength = decodeBytes.length;
+
+        //first byte is version
```

Comment:
If you're doing this, you might want to consider generating the Java classes from [https://github.com/google/certificate-transparency/blob/master/proto/ct.proto](https://github.com/google/certificate-transparency/blob/master/proto/ct.proto )